snapcraft/_store.py

```diff
@@ -190,6 +190,41 @@ def register(snap_name, is_private=False):
         snap_name))
 
 
+def sign_build(snap_filename, key_name='default', local=False):
+    if not repo.is_package_installed('snapd'):
+        raise EnvironmentError(
+            'The snapd package is not installed. In order to use '
+            '`sign-build`, you must run `apt install snapd`.')
+
+    if not os.path.exists(snap_filename):
+        raise FileNotFoundError(
+            'The file {!r} does not exist.'.format(snap_filename))
+
+    snap_series = storeapi.constants.DEFAULT_SERIES
+    snap_yaml = _get_data_from_snap_file(snap_filename)
+    snap_name = snap_yaml['name']
+
+    # test snap has no grade, and so other old snaps that
+    # might need to be asserted once snap-build is supported
+    grade = snap_yaml.get('grade', 'stable')
+
+    store = storeapi.StoreClient()
+    with _requires_login():
+        try:
+            info = store.get_account_information()
+            authority_id = info['account_id']
+            snap_id = info['snaps'][snap_series][snap_name]['snap-id']
+        except KeyError:
+            raise RuntimeError(
+                'Your account lacks information to assert this snap.')
```

Comment:
In various places in this PR you talk about asserting a snap, asserting a built snap file, and asserting a build.  I think it would be worth settling on a consistent vocabulary.